Crash report: VerCors crashed near this position. Cause follows:

# Crash Message
```
VerCors crashed near this position. Cause follows:
  at vct.result.VerificationError$.withContext(VerificationError.scala:71)
  at vct.col.rewrite.NonLatchingRewriter.dispatch(NonLatchingRewriter.scala:12)
  at vct.col.ast.ops.rewrite.VerificationContextRewrite.rewrite(VerificationContextRewrite.scala:6)
  at vct.col.ast.ops.rewrite.VerificationContextRewrite.rewrite$(VerificationContextRewrite.scala:4)
  at vct.col.ast.VerificationContext.rewrite(Node.scala:100)
  at vct.col.ast.ops.rewrite.VerificationContextRewrite.rewriteDefault(VerificationContextRewrite.scala:3)
  at vct.col.ast.ops.rewrite.VerificationContextRewrite.rewriteDefault$(VerificationContextRewrite.scala:3)
  at vct.col.ast.VerificationContext.rewriteDefault(Node.scala:100)
  at vct.col.ast.rewrite.BaseNonLatchingRewriter.dispatch(BaseNonLatchingRewriter.scala:4)
  at vct.col.ast.rewrite.BaseNonLatchingRewriter.dispatch$(BaseNonLatchingRewriter.scala:4)
  at vct.col.rewrite.NonLatchingRewriter.dispatch(NonLatchingRewriter.scala:8)
  at vct.col.ast.ops.rewrite.VerificationRewrite.$anonfun$rewrite$1(VerificationRewrite.scala:6)
  ...
  at vct.main.Main$.main(Main.scala:51)
  at vct.main.Main.main(Main.scala)
```

# Version Information
* VerCors version `9999.9.9-SNAPSHOT`
* At commit be71f0b4c87342ed113d1d684d748e822dea6b0a from branch `improve-extract` (changes=true)

# Arguments
`/home/lars/data/thesis/build/blur.pvl` `--dev-no-sat` `--dev-total-timeout=3600` `--backend-file-base` `/home/lars/data/thesis/build/blur`

# File Inputs
<details>
<summary>blur.pvl</summary>

```pvl

/* MACHINE GENERATED By Halide. */

 decreases;
pure int max(int x, int y) = x > y ? x : y;
 
 decreases;
pure int min(int x, int y) = x > y ? y : x;
 
 decreases;
pure rational max(rational x, rational y) = x > y ? x : y;
 
 decreases;
pure rational min(rational x, rational y) = x > y ? y : x;
 
 decreases;
pure int abs(int x) = x >= 0 ? x : -x;

 decreases;
pure rational abs(rational x) = x >= 0 ? x : -x;
 
 decreases;
pure rational ceil_f32(rational x) = is_int(x) ? x : int_to_rational(rational_to_int(x) + 1);

prover_function int rational_to_int(rational r) \smtlib `(_ to_int)`;

prover_function int int_to_rational(int i) \smtlib `(_ to_real)`;
 
prover_function bool is_int(rational r) \smtlib `(_ is_int)`;
 
prover_function rational div_rat_(rational l, rational r) \smtlib `(_ /)`;
 
prover_function rational pow_f32(rational l, rational r) \smtlib `(_ ^)`;
 
 decreases;
pure rational div_rat(rational l, rational r) = div_rat_(l, r);
 
 decreases;
pure rational fast_inverse_f32(rational x) = div_rat(1.0f, x);
 
 decreases;
pure int round_f32(rational x) = is_int(x) ? rational_to_int(x) : rational_to_int(x + 0.5f);

 decreases;
pure int hdiv(int a, int b) = b==0 ? 0 : a/b;

 decreases;
pure int hmod(int a, int b) = b==0 ? 0 : a%b;

 decreases;
pure int inp_min_0();
 decreases;
pure int inp_min_1();
 decreases;
pure int inp_extent_0();
 decreases;
pure int inp_extent_1();
 decreases;
pure int inp_stride_0() = 1;
 decreases;
pure int inp_stride_1();

 decreases;
pure int inp(int x0, int x1);

 decreases;
pure int blur_y_min_0();
 decreases;
pure int blur_y_min_1();
 decreases;
pure int blur_y_extent_0();
 decreases;
pure int blur_y_extent_1();
 decreases;
pure int blur_y_stride_0() = 1;
 decreases;
pure int blur_y_stride_1();

 ensures \result == hdiv((inp_im(x, y) + inp_im(x + 1, y) + inp_im(x + 2, y)), 3);
 decreases;
pure int blur_x(int x, int y) = hdiv((inp_im(x, y) + inp_im(x + 1, y) + inp_im(x + 2, y)), 3);

 ensures \result == hdiv((hdiv((inp_im(x, y) + inp_im(x + 1, y) + inp_im(x + 2, y)), 3) + hdiv((inp_im(x, y + 1) + inp_im(x + 1, y + 1) + inp_im(x + 2, y + 1)), 3) + hdiv((inp_im(x, y + 2) + inp_im(x + 1, y + 2) + inp_im(x + 2, y + 2)), 3)), 3);
 decreases;
pure int blur_y(int x, int y) = hdiv((blur_x(x, y) + blur_x(x, y + 1) + blur_x(x, y + 2)), 3);

 decreases;
pure int inp_im(int x0, int x1) = inp(x0, x1);
pure int nx(); pure int ny(); 
 requires inp_min_0() == 0 && inp_extent_0() == nx() + 2 && inp_min_1() == 0 && inp_extent_1() == ny() + 2 && inp_stride_1() == nx() + 2 && blur_y_min_0() == 0 && blur_y_extent_0() == nx() && blur_y_min_1() == 0 && blur_y_extent_1() == ny() && blur_y_stride_1() == nx();
 ensures (\forall int x,  int y; 0 <= x && x < nx() && 0 <= y && y < ny(); {:blur_y(x, y):} == hdiv((hdiv((inp_im(x, y) + inp_im(x + 1, y) + inp_im(x + 2, y)), 3) + hdiv((inp_im(x, y + 1) + inp_im(x + 1, y + 1) + inp_im(x + 2, y + 1)), 3) + hdiv((inp_im(x, y + 2) + inp_im(x + 1, y + 2) + inp_im(x + 2, y + 2)), 3)), 3));
 ensures (\forall int x,  int y; blur_y_min_0() <= x && x < (blur_y_min_0() + blur_y_extent_0()) && blur_y_min_1() <= y && y < (blur_y_min_1() + blur_y_extent_1()); {:blur_y(x, y):} == hdiv((hdiv((inp_im(x, y) + inp_im(x + 1, y) + inp_im(x + 2, y)), 3) + hdiv((inp_im(x, y + 1) + inp_im(x + 1, y + 1) + inp_im(x + 2, y + 1)), 3) + hdiv((inp_im(x, y + 2) + inp_im(x + 1, y + 2) + inp_im(x + 2, y + 2)), 3)), 3));
void pipeline(){ }

```
</details>


# Full Log
<details>

```
12:32:18.139 [INFO] Start: VerCors (at 12:32:18)
12:32:18.177 [WARN] Caching is enabled, but results will be discarded, since there were uncommitted changes at compilation time.
12:32:23.784 [INFO] Done: VerCors (at 12:32:23, duration: 00:00:05)
12:32:23.789 [ERROR] vct.result.VerificationError$Crash: VerCors crashed near this position. Cause follows:
	at vct.result.VerificationError$.withContext(VerificationError.scala:71)
	at vct.col.rewrite.NonLatchingRewriter.dispatch(NonLatchingRewriter.scala:12)
	at vct.col.ast.ops.rewrite.VerificationContextRewrite.rewrite(VerificationContextRewrite.scala:6)
	at vct.col.ast.ops.rewrite.VerificationContextRewrite.rewrite$(VerificationContextRewrite.scala:4)
	at vct.col.ast.VerificationContext.rewrite(Node.scala:100)

--- a/src/col/vct/col/ast/util/ExpressionEqualityCheck.scala
+++ b/src/col/vct/col/ast/util/ExpressionEqualityCheck.scala
@@ -1399,8 +1399,13 @@ class AnnotationVariableInfoGetter[G](
       // Collect all vars that are greater than
       val greaterVars: mutable.Set[SymbolicTerm[G]] = mutable.Set()
       vars.foreach { v =>
-        lessThanEqVars.get(v).toSeq.flatMap(_.map(variableSynonyms(_)))
-          .foreach { i => greaterVars.addAll(synonymSets(i)) }
+        lessThanEqVars.get(v).foreach(ltSet =>
+          ltSet.foreach(lt =>
+            variableSynonyms.get(lt).foreach { i =>
+              greaterVars.addAll(synonymSets(i))
+            }
+          )
+        )
       }
       // Redistribute all greater vars again
       vars.foreach { lessThanEqVars.get(_).foreach(_.addAll(greaterVars)) }

